api/src/main/java/me/lokka30/treasury/api/economy/EconomyProvider.java

```diff
@@ -315,10 +316,24 @@ default void retrieveAllAccountsPlayerHasPermission(
                     if (!(account instanceof NonPlayerAccount)) {
                         return CompletableFuture.completedFuture(false);
                     }
-                    return EconomySubscriber.asFuture(subscriber -> ((NonPlayerAccount) account).hasPermission(playerId,
-                            subscriber,
-                            permissions
-                    ));
+
+                    CompletableFuture<Boolean> ret1 = new CompletableFuture<>();
+                    account.hasPermission(playerId, new EconomySubscriber<TriState>() {
+                        @Override
+                        public void succeed(@NotNull final TriState triState) {
+                            if (triState == TriState.UNSPECIFIED || triState == TriState.FALSE) {
+                                ret1.complete(false);
+                                return;
+                            }
+                            ret1.complete(true);
```

Comment:
cant we do
```Java
ret1.complete(triState == TriState.TRUE);
```
?

Reply:
Yes. @lokka30 make this method the one liner mrnemo said

Reply:
resolved :smiley: